lib/active_model/serializer/attributes.rb

```diff
@@ -14,10 +14,12 @@ module Attributes
 
         # Return the +attributes+ of +object+ as presented
         # by the serializer.
-        def attributes(requested_attrs = nil, reload = false)
+        def attributes(options = {}, reload = false)
+          requested_attrs = options[:only]
+          excepts = Array(options[:except])
```

Comment:
Sure amr

Could you share your ams vesion from the gemfile.lock, your serializer,
your render command, and your desired output?

On Mon, Mar 7, 2016 at 10:21 AM Amr Noman <notifications@github.com> wrote:

> In lib/active_model/serializer/attributes.rb
> <https://github.com/rails-api/active_model_serializers/pull/1538#discussion_r55227776>
> :
>
> > @@ -14,10 +14,12 @@ module Attributes
> >
> >          # Return the +attributes+ of +object+ as presented
> >          # by the serializer.
> > -        def attributes(requested_attrs = nil, reload = false)
> > +        def attributes(options = {}, reload = false)
> > +          requested_attrs = options[:only]
> > +          excepts = Array(options[:except])
>
> @bf4 <https://github.com/bf4> I was really looking for something like
> this and I have no idea how to achieve it otherwise, could you please show
> an example of how to do it using attribute unless and instance_options
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails-api/active_model_serializers/pull/1538/files#r55227776>
> .
>